File: algorithms/554. Brick Wall/solution.cc
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        unordered_map<int, int> counter;
        for ( auto & row : wall ) {
            for ( int i = 0, sum = 0; i < row.size() - 1; ++i ) {
                sum += row[i];
                ++counter[sum];
            }
        }
        int maximum = 0;
        for ( auto & c : counter ) {
            maximum = max(maximum, c.second);
        }
        return wall.size() - maximum;
    }
};

// 85 / 85 test cases passed.
// Status: Accepted
// Runtime: 36 ms

File: algorithms/554. Brick Wall/solution.py
class Solution(object):
    def leastBricks(self, wall):
        """
        :type wall: List[List[int]]
        :rtype: int
        """
        counter = collections.defaultdict(int)
        for row in wall:
            s = 0
            for j in xrange(len(row) - 1):
                s += row[j]
                counter[s] += 1
        maximum = max(counter.itervalues()) if counter else 0
        return len(wall) - maximum

# 85 / 85 test cases passed.
# Status: Accepted
# Runtime: 105 ms
